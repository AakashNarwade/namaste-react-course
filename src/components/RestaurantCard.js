import React from "react";
import { CDN_URL } from "../utils/constants";
const RestaurantCard = ({ resData }) => {
  // console.log(resData);
  const {
    name,
    costForTwo,
    avgRating,
    locality,
    areaName,

    cuisines,
    cloudinaryImageId,
  } = resData?.info;
  // console.log(name, costForTwo, avgRating);
  return (
    <div className="restaurant-card">
      <img
        className="restaurant-logo"
        src={CDN_URL + cloudinaryImageId}
        alt="restaurant-logo"
      />
      {/* <img
          className="restaurant-logo"
          src="https://media-assets.swiggy.com/swiggy/image/upload/fl_lossy,f_auto,q_auto,w_660/srwspnhuccrtyr02gnty"
          alt="restaurant-logo"
        /> */}
      <h3 className="restaurant-name">{name}</h3>
      {/* <h4>{cuisines.join(", ")}</h4> */}
      <h4>{`${locality} , ${areaName}`}</h4>
      <h4>{avgRating} stars</h4>
      <h4>{costForTwo}</h4>
      <h4>{resData?.info?.sla?.deliveryTime} minutes</h4>
    </div>
  );
};

export default RestaurantCard;
