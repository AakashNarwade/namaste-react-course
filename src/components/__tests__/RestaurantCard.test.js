import RestaurantCard from "../RestaurantCard";
import MOCK_DATA from "../../components/mocks/resCardMock.json";
import { screen, render } from "@testing-library/react";
import { withPromotedLabel } from "../RestaurantCard";
import "@testing-library/jest-dom";

it("Should load RestaurantCard Component with props Data", () => {
  render(<RestaurantCard resData={MOCK_DATA} />);
  const name = screen.getByText("Starbucks Coffee");
  expect(name).toBeInTheDocument();
});

// it("Should load RestaurantCard Component with withPromotedLabel", () => {
//   const { baseElement } = render(<RestaurantCard />);
//   console.log(baseElement);
//   //   const name = screen.getByText("withPromotedLabel");
//   expect(baseElement).toMatchSnapshot();
// });
