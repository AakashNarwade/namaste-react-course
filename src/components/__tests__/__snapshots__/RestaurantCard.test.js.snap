// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should load RestaurantCard Component with withPromotedLabel 1`] = `
<body>
  <div />
</body>
`;
